openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -515,6 +521,86 @@ def _set_unusable_password(user):
     user.save()
 
 
+class AccountRetirementPartnerReportView(ViewSet):
+    """
+    Provides API endpoints for managing partner reporting of retired
+    users.
+    """
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, CanRetireUser,)
+    parser_classes = (JSONParser,)
+    serializer_class = UserRetirementStatusSerializer
+
+    def _get_orgs_for_user(self, user):
+        """
+        Returns a set of orgs that the user has enrollments with
+        """
+        orgs = set()
+        for enrollment in user.courseenrollment_set.all():
+            org = enrollment.course.org
+
+            # Org can concievably be blank or this bogus default value
+            if org and org != 'outdated_entry':
+                orgs.add(enrollment.course.org)
+        return orgs
+
+    def retirement_partner_report(self, request):  # pylint: disable=unused-argument
+        """
+        GET /api/user/v1/accounts/retirement_partner_report/
```

Comment:
This is a side-effecting GET since it alters the is_being_processed state of the user.  I wonder if it should be changed to POST with the exact same semantics?

Reply:
I'd pondered that, but was saving the POST for another call. I think that other one can/should actually be a PUT, though, so this should work out. 👍